app/models/submission_match.rb

```diff
@@ -13,20 +13,40 @@ def initialize(params)
 
   def find_all
     submissions =
-      Submission.not_deleted.where(query).includes(:user).order(order_by)
-    submissions = submissions.search(term) if term
-
-    # only show submissions that lack an accepted offer when filtering by assignee plus published or accepted [CX-812]
-    return submissions unless filtering_by_assigned_without_accepted_offer?
+      if filtering_by_assigned_without_accepted_offer?
+        submissions_assigned_without_accepted_offer
+      else
+        Submission
+      end
 
-    submissions.filter do |s|
-      s.partner_submissions.size.zero? ||
-        s.partner_submissions.none? { |ps| ps.accepted_offer.present? }
-    end
+    submissions =
+      submissions.not_deleted.where(query).includes(:user).order(order_by)
+    submissions = submissions.search(term) if term
+    submissions
   end
 
   private
 
+  def submissions_assigned_without_accepted_offer
+    sql = <<SQL.squish
+    WITH wrap AS (
+      SELECT s.id as submission_id,
+             COUNT(DISTINCT ps.id) AS count_partner_submissions,
+             COUNT(DISTINCT ps.accepted_offer_id) AS count_accepted_offers
+      FROM submissions s
+       LEFT JOIN partner_submissions ps on ps.submission_id = s.id
+      GROUP BY 1
+      )
+      SELECT submission_id
+      from wrap
+      where count_partner_submissions = 0 OR count_accepted_offers = 0
```

Comment:
```suggestion
    WITH wrap AS (
      SELECT s.id as submission_id,
             COUNT(DISTINCT ps.id) AS count_partner_submissions,
             COUNT(DISTINCT ps.accepted_offer_id) AS count_accepted_offers
      FROM submissions s
       LEFT JOIN partner_submissions ps on ps.submission_id = s.id
      GROUP BY 1
      )
      SELECT submission_id
      FROM wrap
      WHERE count_partner_submissions = 0 OR count_accepted_offers = 0
```

Non-blocking & not a big deal. I'd imagine we don't have any rubocopping on format of SQL queries, so I think we're on our own to manage these kinds of consistencies.